Standards/scs-0112-v1-sonic.md

```diff
@@ -0,0 +1,108 @@
+
+## Design Considerations
+
+There are different ways SONiC support can be implemented in SCS, very similar to existing approaches with Linux.
+
+### Options considered
+
+#### Option 1: SCS distribution of SONiC
+
+With this approach, SCS will create it's own distribution of SONiC, similar to what Debian or Arch are for Linux. This distribution will be based on the SONiC community distribution, but will have SCS specific modules, which will be developed and maintained by SCS. SCS will contribute its code to dedicated SCS repositories and build its own SONiC images. The code can eventually be pushed upstream, but not as top priority. This approach will allow SCS to have a clear path for SONiC support and integration in SCS, but will also require SCS to maintain a distribution of SONiC, which is a significant effort. Upstream/downstream changes will have to be managed and maintained. However the advantage is that SCS will have full control over the distribution and can make changes as needed. Users will have to use the SCS distribution of SONiC, which will be based on the community distribution. If users already deploy community or enterprise SONiC, a migration path to SCS SONiC will be needed.
+
+#### Option 2: SCS will support SONiC but will not change it
+
+SCS supports enterprise ans community versions of SONiC but will not develop its own code for it. This will significantly limit the ability to develop new features for SDN, because all changes will be done in the tooling around SONiC and not in the OS itself. The advantages are that SCS will still improve SONiC support and will have minimal effort for this. The downside is that some features like OVN control plane for SONiC will not be possible.
+
+#### Option 3: SCS develops SCS-specific modules as add-on for any SONiC (Community or Enterprise)
+
+In option 3, SCS will change SONiC by releasing its own modules for it. Those module can be provided as add-ons and installed on top of any version, community or enterprise. While compatability between the modules the SONiC releases will need to be maintained, there will be much broader support for SONiC and users will be able to pick and chose distributions based on their existing relationships and experience and use SCS independent of this. In cases where SCS provides contributions to core SONiC, those can be made in upstream Community repositories, so that the whole community including the propriatory vendors can adopt them eventually.
+
+#### Option 4: SCS does not adopt SONiC at all
+
+This option entails no dedicated effort on SCS's part in supporting SONiC network equipement for it's users and software stack. Users can still use SONiC from what is available by other projects or if they invest the effort themselves. This has several disadvantages:
```

Comment:
It also has an advantage: Less of a deterrent for people who are *not* leveraging SONiC and are not intending to approach a scale where SONiC would be useful.

(yes, it is a deterrent even if it's not a requirement to use SONiC with SCS, unless it's well documented how not needed it is)

Reply:
I do not see it as clear advantage because I don't see it as "either/or" question about if we use SONiC or not. SCS users should always have the choice, regardless if SCS invest in wider support for it. Fact is, that there are many other network OSs and SONiC is a small piece of what can be used.

Reply:
Of course SCS users can still opt to *not* use SONiC. Still, without clear communication that that's the case, it may be deterring to people evaluating cloud solutions. If it looks as if they need to opt into SONiC to use SCS, it will be deterring.

Which is why I said that it needs clear documentation and I think that's a disadvantage of choosing to extensively support SONiC.

Reply:
I think this is a textbook example of FUD.

Just because you are supporting solution "a", this doesn't logically imply anything regarding other solutions "b" and "c", they might or might not be equally supported. This is a logical fallacy I would expect everyone who evaluates cloud solutions to recognize and expect them to avoid such fallacy.

Reply:
Hi @artificial-intelligence ,

If I understand you correctly, it suggest to not list any disadvantages in Option 4, because this should be out of scope of the ADR. I guess we should decide if to remove option 4 altogether, or only list advantages. I initially didn't include it, because it is a normal thing to consider by the reader, what will happen if we don't have a SONiC strategy.

Reply:
> If we don't want to support anything then we certainly don't need a draft document as a standard for that, because the list of things we don't support is arbitrary long and we don't write down everything we don't support (like running SCS on Amiga OS).

I agree that a standard saying "SCS does not adopt X at all" could be applied to basically every other existing technology. Hence, this **standard** would be completely superfluous.

However, AFAIK, the concept of a **decision record** (as this document is declared as) means that this document is not a standard one could comply to, but a mere record of what decision SCS made regarding a technology. That way, if someone wonders in the future why SONIC is (or is not) used in SCS, they can look at a document outlining the reasoning (at the time) which was accepted by a certain quorum.

So while I do not think it is practicable to do that for every technology anyone ever mentions, a decision record with the result "SCS does not adopt SONiC at all" is fine.

Reply:
> > If we don't want to support anything then we certainly don't need a draft document as a standard for that, because the list of things we don't support is arbitrary long and we don't write down everything we don't support (like running SCS on Amiga OS).
> 
> I agree that a standard saying "SCS does not adopt X at all" could be applied to basically every other existing technology. Hence, this **standard** would be completely superfluous.
> 
> However, AFAIK, the concept of a **decision record** (as this document is declared as) means that this document is not a standard one could comply to, but a mere record of what decision SCS made regarding a technology. That way, if someone wonders in the future why SONIC is (or is not) used in SCS, they can look at a document outlining the reasoning (at the time) which was accepted by a certain quorum.
> 
> So while I do not think it is practicable to do that for every technology anyone ever mentions, a decision record with the result "SCS does not adopt SONiC at all" is fine.

Well, the decision is Option 3. This has been discussed and decided within the SCS community. I think the question we are wondering here is if we should include Option 4 (No SONIC support) in the ADR and list all advantages/disadvantages for this option.

Reply:
> A list of (dis)advantages when evaluating products should contain (dis)advantages special to this product, not (dis)advantages regarding any arbitrary solution.

That sounds like a decent middle ground to me I can get behind.

Reply:
Ah, was not aware of that.

(But, if there is no distinct decision record dealing with the question of "**if**", leaving the "**how**" to this document, including the reasoning for "**if**" here would probably help the reader to understand the trade-offs.)